purchase_requisition_from_rfq/models/purchase_requisition.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class PurchaseRequisition(models.Model):
+    _inherit = "purchase.requisition"
+
+    purchase_ids = fields.One2many(
+        default=lambda self: self._context.get('active_ids', []),
+    )
+    line_ids = fields.One2many(
+        default=lambda self: self._get_line(),
+    )
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form',
+                        toolbar=False, submenu=False):
+        res = super(PurchaseRequisition, self).fields_view_get(
+            view_id=view_id, view_type=view_type,
+            toolbar=toolbar, submenu=submenu)
+        if res and view_type in ['tree', 'form']:
+            orders = self.env['purchase.order'].browse(
+                self._context.get('active_ids', []))
+            order_lines = self.env['purchase.order.line']
+            te = orders.mapped('requisition_id')
+            if any(order.state not in ('draft', 'send', 'to approve')
+                   for order in orders):
+                raise ValidationError(_('The requisition cannot be processed '
+                                        'because the order is confirmed!'))
+            if te:
+                raise ValidationError(_(
+                    'Order: %s, already agreement created!') % ', '.join(
+                    orders.filtered(lambda l: l.requisition_id).mapped('name')
+                    ))
+            if len(orders) > 1:
+                order_origin = orders[0]
+                if any(len(order.order_line) != len(order_origin.order_line)
```

Comment:
We should identify first, what we want to compare, i.e., product, qty, price. Then, we an loop through only once to get the list and then compare it.
Code can be shorter using python codiing.